adapters/readpeak/readpeak.go

```diff
@@ -0,0 +1,180 @@
+package readpeak
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+  
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+  
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Readpeak adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+	  endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+
+	requestCopy := *request
+	var rpExt openrtb_ext.ImpExtReadpeak
+	var imps []openrtb2.Imp
+	for i := 0; i < len(requestCopy.Imp); i++ {
+		if requestCopy.Imp[i].Native == nil && requestCopy.Imp[i].Banner == nil {
+			continue
+		}
+		var impExt adapters.ExtImpBidder
+		err := json.Unmarshal(requestCopy.Imp[i].Ext, &impExt)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		err2 := json.Unmarshal(impExt.Bidder, &rpExt)
+		if err2 != nil {
+			errors = append(errors, err2)
+			continue
+		}
```

Comment:
avoid use of `err2`. could be refactored as below.

```
		if err := json.Unmarshal(impExt.Bidder, &rpExt); err2 != nil {
			errors = append(errors, err)
			continue
		}

```

Reply:
Refactored